<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>P2P 聊天客户端</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QGroupBox" name="connectionGroup">
       <property name="title">
        <string>服务器连接</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>地址:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="serverAddressInput">
          <property name="minimumSize">
           <size>
            <width>120</width>
            <height>0</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>端口:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="serverPortInput">
          <property name="maximumSize">
           <size>
            <width>60</width>
            <height>16777215</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>用户名:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="usernameInput">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>0</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="autoReconnectCheck">
          <property name="text">
           <string>自动重连</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="connectButton">
          <property name="text">
           <string>连接</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="disconnectButton">
          <property name="text">
           <string>断开</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="statusLabel">
       <property name="text">
        <string>未连接</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QGroupBox" name="chatGroup">
       <property name="title">
        <string>聊天</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QTextBrowser" name="chatText">
          <property name="minimumSize">
           <size>
            <width>600</width>
            <height>400</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLineEdit" name="messageInput">
            <property name="placeholderText">
             <string>输入消息... (按Enter发送)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="uploadButton">
            <property name="text">
             <string>上传文件</string>
            </property>
            <!--<property name="icon">
             <iconset>
              <normaloff>:/icons/upload.png</normaloff>:/icons/upload.png</iconset>
            </property>-->
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="sendButton">
            <property name="text">
             <string>发送</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QProgressBar" name="uploadProgressBar">
            <property name="value">
             <number>0</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="uploadStatusLabel">
            <property name="text">
             <string>就绪</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="usersGroup">
       <property name="title">
        <string>在线用户</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QListWidget" name="userList">
          <property name="minimumSize">
           <size>
            <width>200</width>
            <height>0</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QPushButton" name="clearButton">
            <property name="text">
             <string>清空聊天</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="settingsButton">
            <property name="text">
             <string>设置</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
